### PR TITLE
WT-9210 Create directory structure for simulator including CMake files

### PR DESCRIPTION
**Pull request by Siddhartha8899**
The directory structure looks like this:
```
Simulator
|– third_party
|---- nlohmann
|------ json.hpp
|– timestamp
|---- CMakeLists.txt
|---- timestamp_simulator.cpp
CMakeLists.txt
```

I have linked the patch build in the ticket itself.

### CODE REVIEW

**Comment by Siddhartha8899** [comment]
The whole code for the JSON parser is in a single header file [json.hpp](https://github.com/wiredtiger/wiredtiger/pull/8054/files#diff-d2e9054fecf15af1c6f3d0a2b50190843611b4da9a0ab940e9253e6ee9ce6d43).

File: test/simulator/timestamp/timestamp_simulator.cpp [hunk 1]
```diff
@@ -0,0 +1,13 @@
+#include <iostream>
```
Comment (tammybailey) [inline test/simulator/timestamp/timestamp_simulator.cpp:1]: I believe we need to add the copyright header on cpp files.

**Review by tammybailey — APPROVED** [review]
LGTM other than my one comment

File: test/simulator/timestamp/timestamp_simulator.cpp (hunk 1 above)
Comment (Siddhartha8899) [reply test/simulator/timestamp/timestamp_simulator.cpp:29]: Updated in [13e01a3](https://github.com/wiredtiger/wiredtiger/pull/8054/commits/13e01a31a3fa34556a2c37369139bf723852fc54)

**Review by bvpvamsikrishna — APPROVED** [review]
lgtm

**Comment by Sean04** [comment]
LGTM

**Review by Sean04 — APPROVED** [review]

**Pull request merged**